src/erlcloud_kinesis.erl

```diff
@@ -560,3 +564,117 @@ split_shards(StreamName, ShardToSplit, NewStartingHashKey) ->
+add_tags(StreamName, Tags) ->
+    add_tags(StreamName, Tags, default_config()).
+
+-spec add_tags(binary(), [{binary(), binary()}], aws_config()) ->
+    ok | {error, any()}.
+add_tags(StreamName, Tags, Config) when is_record(Config, aws_config) ->
+    Json      = [{<<"StreamName">>, StreamName},
+                 {<<"Tags">>,       Tags}],
+    Operation = "Kinesis_20131202.AddTagsToStream",
+    Response  = erlcloud_kinesis_impl:request(Config, Operation, Json),
+    normalize_tags_response(Response).
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Kinesis API: ListTagsForStream
+%% Lists the tags for the specified Amazon Kinesis stream.
+%%
+%% erlcloud_kinesis:list_tags(<<"stream_name">>, <<"key2">>, 2).
+%% {ok, [
+%%       {<<"HasMoreTags">>, true},
+%%       {<<"Tags">>,        [[{<<"Key">>, <<"k1">>}, {<<"Value">>, <<"v1">>}],
+%%                            [{<<"Key">>, <<"k2">>}, {<<"Value">>, <<"v2">>}]]}
+%%      ]
+%% }.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+-spec list_tags(binary()) -> {ok, proplist()} | {error, any()}.
+list_tags(StreamName) ->
+    list_tags(StreamName, default_config()).
+
+-spec list_tags(binary(), aws_config()) -> {ok, proplist()} | {error, any()}.
+list_tags(StreamName, Config) when is_record(Config, aws_config) ->
+    list_tags(StreamName, undefined, undefined, Config).
+
+-spec list_tags(binary(), binary(), integer()) ->
+    {ok, proplist()} | {error, any()}.
+list_tags(StreamName, ESK, Limit) ->
+    list_tags(StreamName, ESK, Limit, default_config()).
+
+-spec list_tags(binary(),
+                binary() | undefined,
+                integer() | undefined,
+                aws_config()) ->
+    {ok, proplist()} | {error, any()}.
+list_tags(StreamName, ESK, Limit, Config) when is_record(Config, aws_config) ->
+    Json      = [{<<"StreamName">>,           StreamName},
+                 {<<"ExclusiveStartTagKey">>, ESK},
+                 {<<"Limit">>,                Limit}],
+    JsonUpd   = lists:filter(fun filter_undefined_params/1, Json),
+    Operation = "Kinesis_20131202.ListTagsForStream",
+    erlcloud_kinesis_impl:request(Config, Operation, JsonUpd).
+
+filter_undefined_params({_, undefined}) -> false;
+filter_undefined_params(_)              -> true.
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Wrapper around 'list_tags' to return {ok, <list_of_all_tags>}.
+%%
+%% erlcloud_kinesis:list_all_tags(<<"stream_name">>).
+%% {ok, [{<<"k1">>, <<"v1">>},
+%%       {<<"k2">>, <<"v2">>}]
+%% }.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+-spec list_all_tags(binary()) -> {ok, proplist()} | {error, any()}.
+list_all_tags(StreamName) ->
+    case erlcloud_kinesis:list_tags(StreamName) of
+        {ok, [_HasMoreTags, {<<"Tags">>, Tags}]} ->
```

Comment:
don't you need a recursive call when `HasMoreTags` is `true`? This looks broken.